src/Core/Layer/Collector/PackageNameCollector.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Qossmic\Deptrac\Core\Layer\Collector;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+use Qossmic\Deptrac\Contract\Ast\CouldNotParseFileException;
+use Qossmic\Deptrac\Contract\Ast\TokenReferenceInterface;
+use Qossmic\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeReference;
+use Qossmic\Deptrac\Core\Ast\Parser\NikicPhpParser\NikicPhpParser;
+
+class PackageNameCollector extends RegexCollector
+{
+    private readonly Lexer $lexer;
+    private readonly PhpDocParser $docParser;
+
+    public function __construct(
+        private readonly NikicPhpParser $nikicPhpParser
+    ) {
+        $this->lexer = new Lexer();
+        $this->docParser = new PhpDocParser(new TypeParser(), new ConstExprParser());
+    }
+
+    public function satisfy(array $config, TokenReferenceInterface $reference): bool
+    {
+        if (!$reference instanceof ClassLikeReference) {
```

Comment:
Limiting this only to class-likes seems half done looking at the docs for that annotation: https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/package.html#description